zst-volatile-macro/src/lib.rs

```diff
@@ -1,19 +1,23 @@
 use proc_macro::TokenStream;
 use quote::{format_ident, quote};
-use syn::{parse_macro_input, Attribute, Data, DeriveInput, Error};
+use syn::{
+    parse_macro_input, punctuated::Punctuated, Attribute, Data, DeriveInput, Error, Lit, Meta,
+    NestedMeta, Token,
+};
 
 #[proc_macro_derive(VolatileStruct)]
 pub fn derive_volatile(tts: TokenStream) -> TokenStream {
-    // FIXME: Check for repr(C), currently we just assume this.
-
     let input = parse_macro_input!(tts as DeriveInput);
 
-    let packed = parse_repr(&input.attrs).unwrap();
-
     let Data::Struct(strukt) = &input.data else {
         return Error::new_spanned(&input,"only structs are supported").into_compile_error().into();
     };
 
+    let (repr_c, repr_align, repr_packed) = parse_repr(&input.attrs).unwrap();
+    if !repr_c {
+        panic!("VolatileStruct needs #[repr(C)]")
+    }
```

Comment:
Let's return an `syn::Error` here. This has the advantage that we can let the user know which part of their code was unexpected. Panicking/unwrapping in proc-macros in generally bad practice because the resulting compiler error messages don't provide much information.

Reply:
Got it, will change it in the next commit.